### PR TITLE
[ETCM-437] Gas estimate for advanced tx

### PR DESCRIPTION
**Pull request by conqeror**

### CODE REVIEW

File: src/wallets/sendTransaction/SendAdvancedTransaction.tsx [hunk 1]
```diff
@@ -16,33 +17,62 @@ import {
...
 
+  const [gasEstimate, setGasEstimate] = useState<undefined | number>(undefined)
+
+  useEffect(() => {
+    async function doGasEstimate(): Promise<void> {
+      try {
+        const estimatedGas = await estimateGas(getSendTransactionParams(transactionParams))
+        setGasEstimate(estimatedGas)
+      } catch {
+        setGasEstimate(undefined)
```
Comment (kapke) [inline src/wallets/sendTransaction/SendAdvancedTransaction.tsx:58]: and log error?

File: src/wallets/sendTransaction/SendAdvancedTransaction.tsx [hunk 2]
```diff
@@ -16,33 +17,64 @@ import {
...
   const hexValidator = createHexValidator(t)
 
+  const feeLimit = new BigNumber(gasLimit).times(gasPrice)
+  const totalAmount = asEther(new BigNumber(amount).plus(feeLimit))
+
+  const remainingBalance = asWei(
+    totalAmount.isFinite() ? availableAmount.minus(totalAmount) : availableAmount,
+  )
+
+  const disableSend = !remainingBalance.isPositive()
```
Comment (qwhex) [inline src/wallets/sendTransaction/SendAdvancedTransaction.tsx]: `!x.isZero() && x.isNegative()` ?

**Review by qwhex — COMMENTED** [review]
Advanced send tx still allows to continue with blank amount, was it planned to be fixed?
Otherwise LGTM!

File: src/wallets/sendTransaction/SendAdvancedTransaction.tsx (hunk 2 above)
Comment (conqeror) [reply src/wallets/sendTransaction/SendAdvancedTransaction.tsx]: Isn't it the same as simply `x.isNegative()`?

Comment (qwhex) [reply src/wallets/sendTransaction/SendAdvancedTransaction.tsx]: It's not the same for `-0` (yes). I'm not sure though if it's possible to get that value here.
It should be ok to simply use `isNegative`.

**Comment by kapke** [comment]
There's couple of thing I find not quite nice here, they're about UX though, not code: 
  - there's no tx cost estimate in ETC so when one puts to high value into gas price/limit and send button is deactivated - one can only guess why was that
  - I was a bit confused about gas limit/estimated gas a bit at first - I haven't noticed the estimated value didn't populate the field that's actually used, they also look basically the same which makes things bit harder to notice IMO
  
Side note - I couldn't create a smart contract, not sure where the issue arises from now.

File: src/wallets/sendTransaction/SendAdvancedTransaction.tsx (hunk 2 above)
Comment (conqeror) [reply src/wallets/sendTransaction/SendAdvancedTransaction.tsx]: I see, we can do `x.isLessThan(0)` then which covers the case with negative zero.

**Comment by conqeror** [comment]
@kapke I added error when funds are insufficient. Regarding your second note and UX in general, I completely agree. I wrote to Russ asking him to do a walkthrough for this whole advanced tx flow together so that we can align on how it should look like from the UX perspective :)

**Review by kapke — APPROVED** [review]
LGTM!

**Review by qwhex — APPROVED** [review]
:shipit:

**Pull request merged**